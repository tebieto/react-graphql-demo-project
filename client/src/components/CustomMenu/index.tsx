import React from 'react';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import { Menu, MenuItem } from '@material-ui/core';
import colors from '../../utils/colors';
const CustomMenu = (): JSX.Element => {
  const target = document.getElementById('null');
  const [anchorEl, setAnchorEl] = React.useState(target);
  const handleClick = React.useCallback(
    (e: React.SyntheticEvent<HTMLElement>) => {
      setAnchorEl(e.currentTarget);
    },
    [],
  );

  const handleClose = React.useCallback(() => {
    setAnchorEl(null);
  }, []);

  const handleLogout = React.useCallback(() => {
    window.location.reload();
  }, []);

  const id = 'custom-menu';

  return (
    <div>
      <div
        className="icon"
        aria-controls={id}
        aria-haspopup="true"
        onClick={handleClick}
      >
        <ExpandMoreIcon style={{ cursor: 'pointer' }} />
      </div>
      <Menu
        id={id}
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem
          onClick={handleLogout}
          style={{ color: colors.logoutButtonColor }}
        >
          Logout
        </MenuItem>
      </Menu>
    </div>
  );
};

export default CustomMenu;
