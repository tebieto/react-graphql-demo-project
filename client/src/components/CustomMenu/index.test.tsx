import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomMenu from '.';

it('renders correctly', () => {
  const tree = shallow(<CustomMenu />);
  expect(toJson(tree)).toMatchSnapshot();
});
