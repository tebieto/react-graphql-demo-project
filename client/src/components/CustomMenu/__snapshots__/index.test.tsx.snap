// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    aria-controls="custom-menu"
    aria-haspopup="true"
    className="icon"
    onClick={[Function]}
  >
    <Memo(ForwardRef(ExpandMoreIcon))
      style={
        Object {
          "cursor": "pointer",
        }
      }
    />
  </div>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    id="custom-menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
      style={
        Object {
          "color": "red",
        }
      }
    >
      Logout
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</div>
`;
